### ISSUE TITLE
Patcher Skipping Most NPCs

### ISSUE THREAD

**Issue by Gawdzila** [Synthesis-Collective/SynthEBD#112]
I'm a bit puzzled by this one, since I changed literally nothing between one patcher run and the next.  I don't think I even closed SynthEBD.  All I did was delete the consistency of one NPC (Nilsine Shatter-Shield) and re-run the patcher.

I had been using 0.9.7.1, so I thought maybe the license had silently expired or something, and I updated to 0.9.9 to see if any of the new changes would fix it.  Unfortunately there seems to be the same issue: the patcher will run extremely quickly, finishing in around ~20 seconds instead of the usual; ~50-60, and the report will show very low NPC counts.  34 Nord Males?  17 Nord Females?  There are no errors, but the deleted consistency for Nilsine has also not been replaced by the patcher, so it seems like she hasn't been processed.

As I said, the behavior began suddenly; I can't think of anything that I changed.
Here's the log, let me know what more specific information you might need.

```
Applying fixed EBD script (for VR or SSE < 1.5.97)
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer IFD Lydia] | WW42LydiaIFDDummy | 31D320:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Rumarin] | WW42RumarinDummy | 2C71A8:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Triss] | WW42TrissDummy | 3AAFD0:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Kaidan] | WW42KaidanDummy | 164A6B:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Lucien] | WW42LucienDummy | 164A6A:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Sofia] | WW42SofiaDummy | 0E6124:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Mirai] | WW42MiraiDummy | 164A69:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Mrissi] | WW42MrissiDummy | 4248DA:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Inigo] | WW42InigoDummy | 0C7AF1:ImGladYoureHere.esp
No valid asset packs could be found for NPC Report me as a bug if you see me in game. [Refer Auri] | WW42AuriDummy | 0D6E02:ImGladYoureHere.esp
No valid asset packs could be found for NPC Nubaree | _NubareeRDO | CAA801:Relationship Dialogue Overhaul.esp
No valid asset packs could be found for NPC Razita | _RazitaRDO | CBEC0D:Relationship Dialogue Overhaul.esp
No valid asset packs could be found for NPC Akisha Mim-La | RDBAkishaMimLa | 320857:DBReformation.esp
No valid asset packs could be found for NPC Hasirr Ahoni | RDBHasirrAhoni | 343FB3:DBReformation.esp
No valid asset packs could be found for NPC  | dunFellglow_TreasCorpseCommonerKhajiitMalePrisoner | 10C687:Skyrim.esm
No valid asset packs could be found for NPC  | WEAdventurerMageShieldStaffArgonianMStorm | 10612F:Skyrim.esm
No valid asset packs could be found for NPC  | WEAdventurerMageShieldStaffArgonianMIce | 10612E:Skyrim.esm
No valid asset packs could be found for NPC  | WEAdventurerMageShieldStaffArgonianMFire | 10612D:Skyrim.esm
No valid asset packs could be found for NPC  | WEAdventurerBattlemageKhajiitMStorm | 106129:Skyrim.esm
No valid asset packs could be found for NPC  | WEAdventurerBattlemageKhajiitMIce | 106128:Skyrim.esm
No valid asset packs could be found for NPC  | WEAdventurerBattlemageKhajiitMFire | 106127:Skyrim.esm
No valid asset packs could be found for NPC Watches-The-Roots | WatchesTheRootsCorpse | 04C757:Skyrim.esm
Finished patching in 00:00:21
Writing combination log to I:\Game Mods\Skyrim SE Mods\Tools\SynthEBD\Settings\Logs\2023-03-19-13-54\Generated Combinations.txt
Writing BodySlide Assignments to D:\Bethesda\Skyrim VR\Data\SynthEBD\BodySlideAssignments.json
Primary asset pack assignment counts:
Breton Male: 5 of 5 (100.00%)
Breton Vampire Male: 1 of 1 (100.00%)
DA13Afflicted Male: 5 of 5 (100.00%)
DarkElf Male: 6 of 6 (100.00%)
HighElf Male: 8 of 8 (100.00%)
Imperial Male: 12 of 12 (100.00%)
Khajiit Male: 0 of 5 (0.00%)
Nord Male: 34 of 34 (100.00%)
Orc Male: 3 of 3 (100.00%)
Argonian Female: 0 of 2 (0.00%)
Breton Female: 3 of 3 (100.00%)
DA13Afflicted Female: 2 of 2 (100.00%)
DarkElf Female: 2 of 2 (100.00%)
HighElf Female: 1 of 1 (100.00%)
HighElf Vampire Female: 1 of 1 (100.00%)
Imperial Female: 5 of 5 (100.00%)
Khajiit Female: 0 of 7 (0.00%)
Nord Female: 17 of 17 (100.00%)
Orc Female: 5 of 5 (100.00%)
Redguard Female: 2 of 2 (100.00%)
WoodElf Female: 1 of 1 (100.00%)
Writing output file to D:\Bethesda\Skyrim VR\Data\SynthEBD.esp.
```

**Comment by Piranha91** [comment]
Can you post a screenshot of your General Settings and TexMesh settings pages?

**Comment by Gawdzila** [comment]
Oh, and one more peculiar thing I noticed.  I tried renaming the folder that my patch run was in, so it was no longer just called "SynthEBD".  When I re-ran the patcher, it still overwrote the older file that was in the renamed folder instead of creating a brand new one.  Further, when I deleted the old .esp to force it to write a new one, instead of creating a new SynthEBD folder, it dropped the new patch straight into override.  And there was ONLY a patch, none of the other folders or files that would usually come with it.

Here are the screenshots:

![Annotation 2023-03-19 141730](https://user-images.githubusercontent.com/119834693/226210265-8af6d4da-eab7-47e3-ac94-ea68dd33dde2.png)

![Annotation 2023-03-19 141759](https://user-images.githubusercontent.com/119834693/226210274-849985e5-8551-4cfe-a487-7a3c27549835.png)

**Comment by Gawdzila** [comment]
Here are the environment details:

```
Data Folder: D:\Bethesda\Skyrim VR\Data
Load Order Source: C:\Users\Jason\AppData\Local\Skyrim VR\Plugins.txt
Creation Club Listings: D:\Bethesda\Skyrim VR\Skyrim.ccc
Game Release: SkyrimVR
Load Order: 
(+) Skyrim.esm
(+) Update.esm
(+) Dawnguard.esm
(+) HearthFires.esm
(+) Dragonborn.esm
(+) SkyrimVR.esm
(+) Unofficial Skyrim Special Edition Patch.esp
(+) Vanilla Script MicroOptimizations.esl
(+) Schlongs of Skyrim - Core.esm
(+) FlowerGirls SE.esm
(+) Water for ENB.esm
(+) ApachiiHair.esm
(+) JSwordsD.esm
(+) NorthernIce.esp
(+) High Poly Head.esm
(+) 1Dwemer.esl
(+) 1SilverArmor.esl
(+) 1Markynaz.esl
(+) Ryder_the_Dragonfly.esp
(+) 1NDArmor.esl
(+) ColovianPrince.esl
(+) 1WR.esl
(+) 1NGarmor.esl
(+) FlowerGirls SE Threesome Fix.esl
(+) RSkyrimChildren.esm
(+) Occ_Skyrim_Tamriel.esp
(+) Lightened Skyrim - merged.esp
(+) Landscape and Water Fixes.esp
(+) Skyrim Project Optimization - Full Version.esm
(+) Expressive Facegen Morphs.esl
(+) Halo's Poser.esp
(+) FNISspells.esp
(+) RevertUSSEPFortifyRestoration.esp
(+) UIExtensions.esp
(+) MCMHelper.esp
(+) higgs_vr.esp
(+) FNIS.esp
(+) High Poly Head Vampire Fix.esp
(+) vrik.esp
(+) VrikForceAction.esp
(+) Schlongs of Skyrim.esp
(+) SOS - Smurf Average Addon.esp
(+) CBBE.esp
(+) 3BBB.esp
(+) HIMBO.esp
(+) OBody.esp
(+) SkyUI_SE.esp
(+) RaceMenu.esp
(+) RaceMenuPlugin.esp
(+) RaceMenuMorphsCBBE.esp
(+) RaceMenuMorphsHIMBO.esp
(+) SOSRaceMenu.esp
(+) ShowRaceMenuAlternative.esp
(+) SOSPhysicsManager.esp
(+) ViewYourselfVR.esp
(+) FGAnimationSelector.esp
(+) dD - Enhanced Blood Main.esp
(+) dD-Reduced Splatter Size.esp
(+) dD-Reduced Wound Size.esp
(+) Nature of the Wild Lands.esp
(+) Nature of the Wild Lands - Autumn Birches.esp
(+) WiZkiD Pinewatch.esp
(+) WiZkiD Lund's Hut.esp
(+) WiZkiD Pavo's House.esp
(+) SexLab-AmorousAdventures.esp
(+) FlowerGirls SE - Adventures.esp
(+) FGMoreAdventures.esp
(+) Caught Red Handed - Quest Expansion.esp
(+) Sidequests of Skyrim.esp
(+) sidequests_FGpatch.esp
(+) DBReformation.esp
(+) Innocence Lost - Quest Expansion.esp
(+) Innocence Lost QE - USSEP Patch.esp
(+) SpellWheelVR.esp
(+) Improved Eyes Skyrim.esp
(+) Improved Eyes Skyrim - Vamp.esp
(+) KS Hairdo's.esp
(+) Beards.esp
(+) FreckleMania.esp
(+) FreckleMania2.esp
(+) Lamenthia's Marks of Beauty.esp
(+) TheEyesOfBeauty.esp
(+) SFO_SkinFeatureOverlays.esp
(+) Pocky Punk's Make Up Addon_females.esp
(+) EmbersHD.esp
(+) Mists of Tamriel.esp
(+) StormLightning.esp
(+) Simplicity of Snow.esp
(+) Veydosebrom Regions.esp
(+) Shadows Of Sunlight - In Small Exterior World Spaces.esp
(+) Cathedral - 3D Pine Grass.esp
(+) Landscape Fixes For Grass Mods.esp
(+) Grass FPS Booster.esp
(+) EnhancedLightsandFX.esp
(+) ELFXEnhancer.esp
(+) ELFX Shadows.esp
(+) SMIM-SE-Merged-All.esp
(+) Landscape and Water Fixes - Patch - ELFX Shadow.esp
(+) Landscape and Water Fixes - Patch - LFfGM.esp
(+) Water for ENB (Shades of Skyrim).esp
(+) NAT-ENB.esp
(+) nchardak waterfall fix.esp
(+) Ilinalta.esp
(+) VRWaterColor.esp
(+) HPP - Vaermina's Torpor.esp
(+) GDOS - Splendid Mechanized Dwemer Door.esp
(+) Easy lockpicking.esp
(+) BarenziahQuestMarkers.esp
(+) man_DaedricShrines.esp
(+) BOTI.esp
(+) FSA_FindingSusannaAlive_BotIR.esp
(+) aMidianborn_Skyforge_Weapons.esp
(+) 1SilverArmor.esp
(+) 1Markynaz.esp
(+) 1NDArmor.esp
(+) CRH_USSEP Patch.esp
(+) [COCO] Scarlet Rose.esp
(+) LunarGuardArmor.esp
(+) LunarGuardArsenal_Battleaxe.esp
(+) Asuras Guard [Armor].esp
(+) ColovianNobleClothes.esp
(+) practical_pirate.esp
(+) rugged_rogue.esp
(+) wilderness_witch.esp
(+) [full_inu] Queen Marika's Dress.esp
(+) icebladeofthemonarch.esp
(+) DragonBoneEbonsteel.esp
(+) Yggdrasil.esp
(+) tamu75GreatSword.esp
(+) CL Ebonyblade.esp
(+) CL BladeofSacrifice.esp
(+) CL NettlebaneOrg.esp
(+) JS Unique Utopia SE - Daggers - Johnskyrim.esp
(+) KTABladeOfWoe.esp
(+) AlbionSwords.esp
(+) valkyrja.esp
(+) Dread Knight Weapon Set.esp
(+) DrunkzealotsPracticalFantasyWeapons.esp
(+) FranklyHDImperialArmorsAndWeapons.esp
(+) Vidani's Bag of Holding.esp
(+) FS_UltimateAssortment.esp
(+) DCR_KingCrusaderMegaPack_SSE.esp
(+) KozakowyFalkaArmor.esp
(+) Traveling Mage - My version by Xtudo.esp
(+) Peacekeeper Armour.esp
(+) DX Dark Knight Armor - My Final Version by Xtudo.esp
(+) TrissDress.esp
(+) TheLily.esp
(+) ESO Altmer Armor by NewrMind43.esp
(+) skyforgedShields.esp
(+) skyforgedWeapons.esp
(+) imp_helm_legend.esp
(+) imp_helm_USSEP.esp
(+) lighter pickaxes & woodcutter axes.esp
(+) BlendedRoads.esp
(+) SilverWeapons.esp
(+) _Fuse00_ArmorKellan.esp
(+) ValkyrieSwordShield.esp
(+) Cloaks.esp
(+) Cloaks - Dawnguard.esp
(+) Cloaks - USSEP Patch.esp
(+) Cloaks_SMP_Patch.esp
(+) Rally's Crates.esp
(+) 1nivWICCloaks.esp
(+) ebony_arhchmage_lagrie.esp
(+) GQJ_DG_vampireamuletfix.esp
(+) Circlets and Masks with all Robes and Hoods.esp
(+) Circlets and Masks with all Robes and Hoods - Winter Is Coming.esp
(+) Unique Thane Weapons.esp
(+) NS_HuntingGroundsOutfit.esp
(+) CollegePrestigeMageOutfit.esp
(+) League of Assassins Armor.esp
(+) _Fuse00_ArmorInquisitor.esp
(+) _Fuse00_ArmorTravelingMageCloth.esp
(+) zzMaidOutfits.esp
(+) No NPC Greetings.esp
(+) LessDragons10.esp
(+) Move it Dammit - Less Wait Time and Follow Me Closer.esp
(+) SimpleItemSpawner.esp
(+) KSA_vdk.esp
(+) Perks Unbound - Special Edition.esp
(+) Helmet Perk Remover.esp
(+) InigoPerkPointGiver.esp
(+) UnreadBooksGlow.esp
(+) Toggle Vampire Attacks_SSE.esp
(+) Savage Wolves and Feral Foxes.esp
(+) Modern Brawl Bug Fix.esp
(+) Fluffworks.esp
(+) BetterMagicLights.esp
(+) Better Candlelight.esp
(+) Forceless Dawnbreake Explosion.esp
(+) CampfireFixOptional.esp
(+) EVE_GiantCampfiresCastShadows.esp
(+) BlockingAndFlash.esp
(+) MarketStallsAnim.esp
(+) Relationship Dialogue Overhaul.esp
(+) RichMerchantsSkyrim_x5.esp
(+) MerchantsDontFollow.esp
(+) ImGladYoureHere.esp
(+) Run For Your Lives.esp
(+) Unique Uniques.esp
(+) Trees in Cities.esp
(+) Skyland Imperial Forts and Dungeons.esp
(+) Particle Patch for ENB SSE.esp
(+) Particle Patch for ENB.esp
(+) Unique Character.esp
(+) XPMSE.esp
(+) nwsFollowerFramework.esp
(+) MW Belesister LPsolo.esp
(+) NPC Appearances Merged - Textures0.esp
(+) NPC Appearances Merged.esp
(+) NPC Appearances Merged0.esp
(+) RSChildren.esp
(+) JKJ - Outfit Edits - USSEP.esp
(+) JKJ - Vaharadaro's Jaguar Khayla.esp
(+) Project ja-Kha'jay- Khajiit Diversity.esp
(+) SkyrimMaiqReplacer.esp
(+) COCO Scarlet Rose - Flower Girls SPID.esp
(+) Gabriellawaifu.esp
(+) SPID-Protected Citizens.esp
(+) Mythic Dawn Armor - SPID.esp
(+) Colovian Prince - SPID.esp
(+) LunarGuardSPID.esp
(+) WindRulerSPID.esp
(+) NorthernGodArmorSPID.esp
(+) Dibella Priest marika dress.esp
(+) Traveling Mage Xtudo ver SPID.esp
(+) MaidenNPCs-NoWreaths-SPID.esp
(+) k101Daegon.esp
```

**Comment by Piranha91** [comment]
In your general menu, you don't have an output folder selected so everything is supposed to go to your data (overwrite) folder, no?

**Comment by Gawdzila** [comment]
Ah, yes, you are correct.  I think that got reset when I upgraded to 0.9.9 and I forgot to put that setting back. It now correctly creates a new SynthEBD folder and the other folders and files inside it, and it does assign a BodySlide now, but the patcher still does not assign textures.

Even adding an entry to Specific Assignments did not force the patcher to assign Nilsine any texture assets.  As near as I can tell, the configs are active, textures are allowed, and Nilsine is not on the blocked list anywhere -- not to mention that there are still very few NPCs being given asset packs according to the log.

**Comment by Piranha91** [comment]
If you revert to an earlier version does it work? I can't think of why it wouldn't see your NPCs. Do you have anything in your block list?

**Comment by Gawdzila** [comment]
Ugh, I found the issue. I did have some NPCs in my block list, but they've been there for a long time so I didn't look too hard at it except to look for Nilsine's name.  I also had a couple plugins there too for custom followers.  Unfortunately what I didn't notice was that I managed to put the EasyNPC plugin there.  I don't even know how, maybe an accidental drag-and-drop.  All fixed; closed due to embarassment :)
